### PR TITLE
chore: address comments from PR#873

### PR DESCRIPTION
**Pull request by mabdollahpour-ol**
This PR aims to address the remaining comments from PR#873.

- Generate API documents for modified and new code.
- Make the repository verification check generic.
- Fix repo verification fact parameter docs.

### CODE REVIEW

File: src/macaron/slsa_analyzer/checks/repo_verification_check.py [hunk 1]
```diff
@@ -14,59 +14,63 @@
...
-    repo_link: Mapped[str] = mapped_column(String, nullable=True)
+    #: Repository link identified by deps.dev.
+    deps_dev_repo_link: Mapped[str | None] = mapped_column(
+        String, nullable=True, info={"justification": JustificationType.HREF}
+    )
 
-    # Repository link identified by deps.dev.
-    deps_dev_repo_link: Mapped[str | None] = mapped_column(String, nullable=True)
+    #: Number of stars on the repository identified by deps.dev.
+    deps_dev_stars_count: Mapped[int | None] = mapped_column(
```
Comment (behnazh-w) [inline src/macaron/slsa_analyzer/checks/repo_verification_check.py:40]: I think adding the `deps_dev_` prefix could be confusing. How about using `stars_count` instead?

File: docs/source/index.rst [hunk 2]
```diff
@@ -89,6 +89,9 @@ the requirements that are currently supported by Macaron.
    * - ``mcn_provenance_derived_commit_1``
      - **Provenance derived commit** - Check if the analysis target's commit matches the commit in the provenance.
      - If there is no commit, this check will fail.
+   * - ``mcn_repo_verification_1``
+     - **Source version controlled** - Check if every change to the source is tracked in a version control system.
+     - If the claimed source repository provenance made by a package is not verified, this check will fail. If no claim of a source repository could be found or the build system is not supported, the result of the check will be "unknown".
```
[the remark this answers is not in the document]
Comment (mabdollahpour-ol) [reply docs/source/index.rst:94]: I agree but there is already a "repo_validation" file under "repo_finder" that does a totally different thing.

File: src/macaron/slsa_analyzer/checks/repo_verification_check.py [hunk 3]
```diff
@@ -14,59 +14,63 @@
...
-    group: Mapped[str] = mapped_column(String, nullable=False)
-    artifact: Mapped[str] = mapped_column(String, nullable=False)
-    version: Mapped[str] = mapped_column(String, nullable=False)
+    #: Repository link identified by Macaron's repo finder.
+    repo_link: Mapped[str] = mapped_column(String, nullable=True, info={"justification": JustificationType.HREF})
 
-    # Repository link identified by Macaron's repo finder.
-    repo_link: Mapped[str] = mapped_column(String, nullable=True)
+    #: Repository link identified by deps.dev.
+    deps_dev_repo_link: Mapped[str | None] = mapped_column(
```
[the remark this answers is not in the document]
Comment (mabdollahpour-ol) [reply src/macaron/slsa_analyzer/checks/repo_verification_check.py:35]: Originally I used deps.dev repo to get the stars and forks and as it sometimes have a different result I had to put it there. But I think we can directly use the repo url that we find (repo_link) to get the repo info from deps.dev.

File: docs/source/index.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (behnazh-w) [reply docs/source/index.rst:94]: As discussed offline, that's an internal naming convention, which we can improve for sure. But the check name should be more intuitive. I like your suggestion to use `scm_authenticity_check`.

[the remark this answers is not in the document]
Comment (mabdollahpour-ol) [reply docs/source/index.rst]: @behnazh-w I renamed the check and modified the documentation. Please feel free to review the changes and let me know if any modification is needed.

File: docs/source/index.rst [hunk 4]
```diff
@@ -89,6 +89,9 @@ the requirements that are currently supported by Macaron.
    * - ``mcn_provenance_derived_commit_1``
      - **Provenance derived commit** - Check if the analysis target's commit matches the commit in the provenance.
      - If there is no commit, this check will fail.
+   * - ``mcn_scm_authenticity_check_1``
+     - **Source repo authenticity** - Check whether the claims of a source repository provenance made by a package can be corroborated.
+     - If the source code repository contains conflicting evidence regarding its claim of provenance, this check will fail. If no source repository or corroborating evidence is found, or if the build system is unsupported, the check will return UNKNOWN as the result. This check currently supports only Maven artifacts.
```
Comment (behnazh-w) [inline docs/source/index.rst:94]:
```suggestion
     - **Source repo authenticity** - Check whether the claims of a source code repository made by a package can be corroborated.
     - If the source code repository contains conflicting evidence regarding its claim of the source code repository, this check will fail. If no source code repository or corroborating evidence is found, or if the build system is unsupported, the check will return ``UNKNOWN`` as the result. This check currently supports only Maven artifacts.
```

**Review by behnazh-w — APPROVED** [review]

**Pull request merged**